### PR TITLE
Add sysdeps target

### PR DESCRIPTION
**Pull request by makyo**
This will allow you to bring up a box or a container with the system dependencies required for building the GUI.

### CODE REVIEW

**Comment by makyo** [comment]
# QA

* Bring up a new LXC
* install git and make (depending on your lxc template)
* `git clone https://github.com/makyo/juju-gui.git`
* `cd juju-gui`
* `git checkout sysdeps`
* `sudo make sysdeps`
* Ensure that `make build` completes successfully

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2353/
Test PASSed.

File: vagrant-provision.sh [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
+add-apt-repository -y ppa:chris-lea/node.js
 apt-get update
 apt-get install -y nodejs imagemagick python-sphinx python-yaml \
   python-tz python-virtualenv python-shelltoolbox python-tornado \
   python-gflags g++ xvfb git firefox
 
-npm install -g jshint@2.1.3 mocha-phantomjs@3.3.1 phantomjs@1.9.2-6 #mocha-phantomjs@3.2.0 phantomjs@1.9.1-0
-# mocha-phantomjs@3.3.1 phantomjs@1.9.2-6
\ No newline at end of file
+npm install -g jshint@2.1.3 mocha-phantomjs@3.3.1 phantomjs@1.9.2-6
```
Comment (huwshimi) [inline vagrant-provision.sh]: The way this file is run will result in a `sudo npm install -g ...` which I thought was a bad idea, but I'm not sure if there's anything we an do about it.

**Comment by huwshimi** [comment]
Much nicer, thanks @makyo :+1:

File: vagrant-provision.sh (hunk 1 above)
Comment (hatched) [inline vagrant-provision.sh]: I am running newer versions in my non qa-for-this-lxc so maybe we should update these at some point

**Comment by hatched** [comment]
:+1: QA OK - this is awesome. 
I followed the steps as outlined and then ran the following make commands `prod, devel, test-prod, test-devel` and all worked as intended.

**Comment by hatched** [comment]
I'd probably rename the 'vagrant' file now though

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2354/
Test PASSed.

**Comment by makyo** [comment]
:shipit: Thanks

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**